### PR TITLE
New Trace UI

### PR DESCRIPTION
**Pull request by sprsquish**
This is a re-write of the main trace UI. I'll get an example up soon.

With this re-write all generation happens server side. Once rendered, we use flight to add behavior to the page. The page is rendered as styled divs instead of via d3. The page will now take up the full width of the browser, popups are also larger and clearer. Individual spans can be expanded and collapsed. Filtering of services has been simplified and expanded as well.

### CODE REVIEW

File: zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala [hunk 1]
```diff
@@ -69,34 +68,29 @@ case class Trace(private val s: Seq[Span]) {
...
   /**
    * Get the start and end timestamps for this trace.
    */
   def getStartAndEndTimestamp: Option[Timespan] = {
-    spans.flatMap {
-      s => s.annotations.map {
-        a => a.timestamp
-      }
-    } match {
+    spans.flatMap(_.annotations.map(_.timestamp)) match {
```
Comment (bmdhacks) [inline zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala:91]: If you're going to refactor this, it's begging for a for comprehension

File: zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala [hunk 2]
```diff
@@ -69,34 +68,29 @@ case class Trace(private val s: Seq[Span]) {
...
       case Nil => None // No annotations
-      case s@_ => Some(Timespan(s.min, s.max))
+      case s => Some(Timespan(s.min, s.max))
```
Comment (bmdhacks) [inline zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala:93]: Untyped cases are deprecated

File: zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala [hunk 3]
```diff
@@ -161,11 +155,8 @@ case class Trace(private val s: Seq[Span]) {
   /**
    * Get all the binary annotations in this trace.
    */
-  def getBinaryAnnotations: Seq[BinaryAnnotation] = {
-    spans.map {
-      _.binaryAnnotations
-    }.flatten
-  }
+  def getBinaryAnnotations: Seq[BinaryAnnotation] =
+    spans.map(_.binaryAnnotations).flatten
```
Comment (bmdhacks) [inline zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala:159]: isn't this just flatMap?

File: zipkin-web/src/main/scala/com/twitter/zipkin/web/Main.scala [hunk 4]
```diff
@@ -68,9 +68,9 @@ trait ZipkinWebFactory { self: App =>
       ("/app/", handlePublic(resourceDirs, typesMap, webCacheResources())),
       ("/public/", handlePublic(resourceDirs, typesMap, webCacheResources())),
       ("/", addLayout andThen handleIndex(queryClient)),
-      ("/traces/:id", addLayout(webRootUrl()) andThen handleTraces),
-      ("/static", addLayout(webRootUrl()) andThen handleStatic),
-      ("/aggregates", addLayout(webRootUrl()) andThen handleAggregates),
+      ("/traces/:id", addLayout andThen handleTraces(queryClient)),
+      //("/static", addLayout(webRootUrl()) andThen handleStatic),
+      //("/aggregates", addLayout(webRootUrl()) andThen handleAggregates),
```
Comment (bmdhacks) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Main.scala]: What's your plan for this?

Comment (sprsquish) [inline zipkin-web/src/main/scala/com/twitter/zipkin/web/Main.scala]: As discussed offline, we're working on a better version.

File: zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala (hunk 3 above)
Comment (sprsquish) [inline zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala:159]: yes! :)

File: zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala (hunk 1 above)
Comment (sprsquish) [inline zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala:91]: I think it's clear this way. Not everything needs to be a for comprehension.

File: zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala (hunk 2 above)
Comment (sprsquish) [inline zipkin-common/src/main/scala/com/twitter/zipkin/query/Trace.scala:93]: fixed

**Comment by bmdhacks** [comment]
LGTM

**Comment by sprsquish** [comment]
You can play around with example data in the new UI here: http://zipkin.io/

**Comment by bmdhacks** [comment]
I think I found a bug:
http://zipkin.io/traces/77e8019056bbae52?serviceName=semper
Click Expand All twice.  Now try to expand collapse 'vitae'

**Comment by sprsquish** [comment]
So I can get a sense for what people will expect from this UI, what's the behavior you expect?

**Comment by sprsquish** [comment]
This has been merged internally and will show up here soon.

**Pull request closed without merging**